### PR TITLE
Const was getting dropped from all types, it is now getting added. 

### PR DESCRIPTION
**Pull request by miguelsaldivar**
1. This is a fix for #6186, not only was _const_ getting dropped from `char *`'s but from every type.
    So now `const` is getting added to every type.
2. Here is the output from running `python tests/runner.py test_webidl`:
```bash
Test suites:
['test_core']
Running test_core: (1 tests)
test_webidl (test_core.default) ... (checking sanity from test runner)
INFO:root:(Emscripten: Running sanity checks)
testing mode ALL , memory growth = False
testing mode FAST , memory growth = False
testing mode DEFAULT , memory growth = False
testing mode ALL , memory growth = True
ok

----------------------------------------------------------------------
Ran 1 test in 13.236s

OK
```

### CODE REVIEW

**Comment by kripken** [comment]
Thanks!

Testing this on ammo.js (running `python make.py` there), I get errors with this PR,
```

glue.cpp:76:25: error: cannot initialize a parameter of type 'btCollisionObject *' with an lvalue of type 'const btCollisionObject *'
  self->contactPairTest(arg0, arg1, *arg2);
                        ^~~~
../src/BulletCollision/CollisionDispatch/btCollisionWorld.h:457:42: note: passing argument to parameter 'colObjA' here
        void    contactPairTest(btCollisionObject* colObjA, btCollisionObject* colObjB, ContactResultCallback& resultCallback);
                                                   ^
glue.cpp:80:21: error: cannot initialize a parameter of type 'btCollisionObject *' with an lvalue of type 'const btCollisionObject *'
  self->contactTest(arg0, *arg1);
                    ^~~~
../src/BulletCollision/CollisionDispatch/btCollisionWorld.h:453:38: note: passing argument to parameter 'colObj' here
        void    contactTest(btCollisionObject* colObj, ContactResultCallback& resultCallback);
                                               ^
glue.cpp:84:26: error: cannot initialize a parameter of type 'btCollisionObject *' with an lvalue of type 'const btCollisionObject *'
  self->updateSingleAabb(arg0);
                         ^~~~
../src/BulletCollision/CollisionDispatch/btCollisionWorld.h:148:43: note: passing argument to parameter 'colObj' here
        void    updateSingleAabb(btCollisionObject* colObj);
                                                    ^
glue.cpp:278:25: error: cannot initialize a parameter of type 'btCollisionObject *' with an lvalue of type 'const btCollisionObject *'
  self->contactPairTest(arg0, arg1, *arg2);
                        ^~~~
../src/BulletCollision/CollisionDispatch/btCollisionWorld.h:457:42: note: passing argument to parameter 'colObjA' here
        void    contactPairTest(btCollisionObject* colObjA, btCollisionObject* colObjB, ContactResultCallback& resultCallback);
                                                   ^
glue.cpp:282:21: error: cannot initialize a parameter of type 'btCollisionObject *' with an lvalue of type 'const btCollisionObject *'
  self->contactTest(arg0, *arg1);
                    ^~~~
../src/BulletCollision/CollisionDispatch/btCollisionWorld.h:453:38: note: passing argument to parameter 'colObj' here
        void    contactTest(btCollisionObject* colObj, ContactResultCallback& resultCallback);
                                               ^
glue.cpp:286:26: error: cannot initialize a parameter of type 'btCollisionObject *' with an lvalue of type 'const btCollisionObject *'
  self->updateSingleAabb(arg0);
                         ^~~~
../src/BulletCollision/CollisionDispatch/btCollisionWorld.h:148:43: note: passing argument to parameter 'colObj' here
        void    updateSingleAabb(btCollisionObject* colObj);
                                                    ^
glue.cpp:551:31: error: binding value of type 'const btDefaultVehicleRaycaster::btVehicleRaycasterResult' to reference to type
      'btVehicleRaycaster::btVehicleRaycasterResult' drops 'const' qualifier
  self->castRay(*arg0, *arg1, *arg2);
                              ^~~~~
../src/BulletDynamics/Vehicle/btVehicleRaycaster.h:30:93: note: passing argument to parameter 'result' here
        virtual void* castRay(const btVector3& from,const btVector3& to, btVehicleRaycasterResult& result) = 0;
                                                                                                   ^
glue.cpp:718:25: error: cannot initialize a parameter of type 'btCollisionObject *' with an lvalue of type 'const btCollisionObject *'
  self->contactPairTest(arg0, arg1, *arg2);
                        ^~~~
../src/BulletCollision/CollisionDispatch/btCollisionWorld.h:457:42: note: passing argument to parameter 'colObjA' here
        void    contactPairTest(btCollisionObject* colObjA, btCollisionObject* colObjB, ContactResultCallback& resultCallback);
                                                   ^
glue.cpp:722:21: error: cannot initialize a parameter of type 'btCollisionObject *' with an lvalue of type 'const btCollisionObject *'
  self->contactTest(arg0, *arg1);
                    ^~~~
../src/BulletCollision/CollisionDispatch/btCollisionWorld.h:453:38: note: passing argument to parameter 'colObj' here
        void    contactTest(btCollisionObject* colObj, ContactResultCallback& resultCallback);
                                               ^
glue.cpp:726:26: error: cannot initialize a parameter of type 'btCollisionObject *' with an lvalue of type 'const btCollisionObject *'
  self->updateSingleAabb(arg0);
                         ^~~~
../src/BulletCollision/CollisionDispatch/btCollisionWorld.h:148:43: note: passing argument to parameter 'colObj' here
        void    updateSingleAabb(btCollisionObject* colObj);
                                                    ^
glue.cpp:1603:31: error: binding value of type 'const btDefaultVehicleRaycaster::btVehicleRaycasterResult' to reference to type
      'btVehicleRaycaster::btVehicleRaycasterResult' drops 'const' qualifier
  self->castRay(*arg0, *arg1, *arg2);
                              ^~~~~
../src/BulletDynamics/Vehicle/btRaycastVehicle.h:230:93: note: passing argument to parameter 'result' here
        virtual void* castRay(const btVector3& from,const btVector3& to, btVehicleRaycasterResult& result);
                                                                                                   ^
glue.cpp:1761:10: error: cannot initialize return object of type 'btRigidBody *' with an rvalue of type 'const btRigidBody *'
  return self->upcast(arg0);
         ^~~~~~~~~~~~~~~~~~
glue.cpp:3162:10: error: cannot initialize return object of type 'btSoftBody *' with an rvalue of type 'const btSoftBody *'
  return self->upcast(arg0);
         ^~~~~~~~~~~~~~~~~~
glue.cpp:3794:25: error: cannot initialize a parameter of type 'btCollisionObject *' with an lvalue of type 'const btCollisionObject *'
  self->contactPairTest(arg0, arg1, *arg2);
                        ^~~~
../src/BulletCollision/CollisionDispatch/btCollisionWorld.h:457:42: note: passing argument to parameter 'colObjA' here
        void    contactPairTest(btCollisionObject* colObjA, btCollisionObject* colObjB, ContactResultCallback& resultCallback);
                                                   ^
glue.cpp:3798:21: error: cannot initialize a parameter of type 'btCollisionObject *' with an lvalue of type 'const btCollisionObject *'
  self->contactTest(arg0, *arg1);
                    ^~~~
../src/BulletCollision/CollisionDispatch/btCollisionWorld.h:453:38: note: passing argument to parameter 'colObj' here
        void    contactTest(btCollisionObject* colObj, ContactResultCallback& resultCallback);
                                               ^
glue.cpp:3802:26: error: cannot initialize a parameter of type 'btCollisionObject *' with an lvalue of type 'const btCollisionObject *'
  self->updateSingleAabb(arg0);
                         ^~~~
../src/BulletCollision/CollisionDispatch/btCollisionWorld.h:148:43: note: passing argument to parameter 'colObj' here
        void    updateSingleAabb(btCollisionObject* colObj);
                                                    ^
16 errors generated.
```

**Comment by miguelsaldivar** [comment]
Yeah of course! Okay, I'll look into this when I get home, thanks for the response. :sunglasses:

**Comment by miguelsaldivar** [comment]
@kripken, so the issue was that the ammo.idl did not match some of the Bullet/src files.

File: tools/webidl_binder.py [hunk 1]
```diff
@@ -306,8 +306,13 @@ def base_type_to_c(t):
     return ret
 
   t = t.replace(' (Wrapper)', '')
```
Comment (kripken) [inline tools/webidl_binder.py]: i think maybe all the lines from here to the end could be written with less repetition, like this:
```
prefix = ''
suffix = ''
if '[]' in t:
  t = t.replace('[]', '')
  suffix = '*'
if 'const ' in t:
  t = t.replace('const ', '')
  prefix = 'const '
return prefix + base_type_to_c(t) + suffix
```
What do you think?

Comment (miguelsaldivar) [reply tools/webidl_binder.py]: I agree, that looks a lot cleaner.

File: tools/webidl_binder.py [hunk 2]
```diff
@@ -306,9 +306,16 @@ def base_type_to_c(t):
...
+
+  prefix = ''
+  suffix = ''
   if '[]' in t:
-    return base_type_to_c(t.replace('[]', '')) + '*'
-  return base_type_to_c(t)
+    t = t.replace('[]', '')
+    suffix = '*'
+  if 'const' in t:
```
Comment (kripken) [inline tools/webidl_binder.py]: should be `'const '` (with a space)

**Comment by kripken** [comment]
Great, thanks! One last comment above, and also please add yourself to AUTHORS before we merge.

**Comment by kripken** [comment]
Thanks again! Great work here.

**Pull request merged**